crates/but-cli/src/command/commit.rs

```diff
@@ -21,20 +23,34 @@ pub fn commit(
         .unwrap_or_else(|| Ok(repo.head_id()?))?
         .detach();
 
-    let destination = if amend {
-        if message.is_some() {
-            bail!("Messages aren't used when amending");
-        }
-        but_workspace::commit_engine::Destination::AmendCommit(parent_id)
-    } else {
-        but_workspace::commit_engine::Destination::NewCommit {
-            parent_commit_id: Some(parent_id),
-            message: message.unwrap_or_default().to_owned(),
-            stack_segment_ref: stack_segment_ref.map(|rn| rn.try_into()).transpose()?,
-        }
-    };
     let changes = to_whole_file_diffspec(but_core::diff::worktree_changes(&repo)?.changes);
     if let Some(project) = project.as_ref() {
+        let destination = if amend {
+            if message.is_some() {
+                bail!("Messages aren't used when amending");
+            }
+            but_workspace::commit_engine::Destination::AmendCommit(parent_id)
+        } else {
+            let stack_segment = if let Some(stack_segment_ref) = stack_segment_ref {
+                let full_name = gix::refs::FullName::try_from(stack_segment_ref)?;
+                VirtualBranchesHandle::new(project.gb_dir())
```

Comment:
Thanks for the ping @krlvi!

This will certainly work as a workaround, but also is something that I think must be avoided. Now there is a strong entanglement to V2 and its StackIds.

My intuition here is that disambiguation would need to introduce the stack reference, i.e. the top-most reference in a stack. Somehow I think this underlying problem is also improper validation of the workspace - in theory a single reference should only be assigned to a single stack and not multiple. I added a note about this to https://github.com/gitbutlerapp/gitbutler/pull/7403 and think this can be removed once V3 is more complete.